### ISSUE TITLE
About distributed training of Imagenet, I  am confused there is no operation to collect grads from machines and average them before update grads. 

### ISSUE THREAD

**Issue by TobeyYang** [pytorch/examples#374]
I write a distributed training model refer to the code  imagenet/main.py , and the models on different machine own their independent optimizer. But I noticed that after backward() there is no operation to collect param grads from other processes and average them to get new grads for update. Does pytorch accomplish the average task implicitly by the optimizer.step() function? I am so confused..

**Comment by hszhao** [comment]
@TobeyYang Same question. Do we need to add average_gradients manually after loss.backward() as in https://pytorch.org/tutorials/intermediate/dist_tuto.html

**Comment by JayanthRR** [comment]
@hszhao If you wrap your model using [DistributedDataParallel](https://pytorch.org/docs/stable/nn.html#torch.nn.parallel.DistributedDataParallel) it happens under the hood. The tutorial you are referring to tries to implement the same functionality differently.

**Comment by hszhao** [comment]
@JayanthRR Thanks and got it. Btw, for the current sample code, do you think we need to increase the base lr as 0.2 when using batch size 256 with 2 nodes each 8 gpus?

**Comment by JayanthRR** [comment]
@hszhao I'm sorry, I haven't actually executed this code.
